Source/WebService/Blackjack.WebService/ClientApp/src/app/components/play-game-page/play-game.component.ts

```diff
@@ -27,21 +75,166 @@ export class PlayGameComponent implements OnInit {
    * @param chipValue
    */
   public _addToWager(chipValue: number): void {
+    if (chipValue > this.user.balance) {
+      alert('You have insufficient funds for this wager.');
+      return;
+    }
     this.wager += chipValue;
-    this.player.balance -= chipValue;
+    this.user.balance -= chipValue;
   }
 
   /**
    * Returns funds to the player's balance and resets the wager to 0.
    */
   public _clearWager(): void {
-    this.player.balance += this.wager;
+    this.user.balance += this.wager;
     this.wager = 0;
   }
 
-  public async ngOnInit(): Promise<void> {
-    this.player = await this._userApi.getUser('test');
+  /**
+  * Begins the game by allowing players to enter their name and how many decks to play with.
+  * Also determines if either the player, dealer, or both have black jack.
+  */
+  public startGame(): void {
+    if (isNaN(this.deckQty) || this.deckQty < 1) {
+      alert(`${this.deckQty} is not a valid amount of decks, please try again.`);
+      return;
+    }
+    this.deck.shuffle(this.deckQty);
+    this.player.updateHand(this.deck.deal(2));
+    this.dealer.updateHand(this.deck.deal(2));
+    this.isPregameHidden = true;
+    this.isPlayAreaHidden = false;
+
+    /*Checks for blackjacks. If anyone does, game automatically ends.*/
+    if (this.player.hasBlackJack() && this.dealer.hasBlackJack()) {
+      this.resolveBlackjack(BlackjackOutcomes.BothBlackjack, () => this.pushOutcome());
+      return;
+    }
+    else if (this.player.hasBlackJack() && !this.dealer.hasBlackJack()) {
+      this.resolveBlackjack(BlackjackOutcomes.PlayerBlackjack, () => this.winOutcome());
+      return;
+    }
+    else if (!this.player.hasBlackJack() && this.dealer.hasBlackJack()) {
+      this.resolveBlackjack(BlackjackOutcomes.DealerBlackjack, () => this.loseOutcome());
+      return;
+    }
+    this.nextMove = BlackjackOutcomes.PlayerChoice;
+    this.isGameButtonHidden = false;
+  }
+
+  /**
+   * Gives the player another card but automatically ends play if the player busts. 
+   */
+  public hit(): void {
+    this.player.updateHand(this.deck.deal(1));
+    if (this.player.hasBusted) {
+      this.revealDealer();
+      this.nextMove = BlackjackOutcomes.DealerWin;
+      this.loseOutcome();
+      this.isPlayAgainHidden = false;
+      this.isGameButtonHidden = true;
+    }
+  }
+
+  /**
+   * Ends the players turn, reveals dealer's cards, hides game buttons, and plays
+   * the dealer's hand over two second intervals.
+   */
+  public stay(): void {
+    this.revealDealer();
+    const intervalID = setInterval(() => {
```

Comment:
Throw the following comment over top this line for me

```javascript
// TODO: Use RxJS to perform interval work and move off setInterval
```

setInterval is not incorrect here, but it's one of those things that's good when you're doing vanilla JS work and not using a framework. `setInterval` and `setTimeout` don't always play nice with the Angular framework in some ways, and in some cases don't play well in very annoying ways.

There's a more framework-friendly way to do this with another library that comes packaged with Angular but it's a more intermediate/advanced implementation.

So I want to teach it to you before we consider the project done, but it's not necessary right now.